helm/grafana/templates/cnpg/objectstores.yaml

```diff
@@ -11,11 +11,7 @@ metadata:
 spec:
   configuration:
     destinationPath: {{ .Values.postgresqlCluster.backup.destinationPath | quote }}
-    {{- if .Values.postgresqlCluster.backup.useIRSA }}
-    s3Credentials:
-      inheritFromIAMRole: true
-    {{- end }}
-    {{- if and (.Values.postgresqlCluster.objectStorage.s3.secretAccessKey) (.Values.postgresqlCluster.objectStorage.s3.accessKeyId) }}
+    {{- if and (eq .Values.postgresqlCluster.objectStorage.type "s3") (and (.Values.postgresqlCluster.objectStorage.s3.secretAccessKey) (.Values.postgresqlCluster.objectStorage.s3.accessKeyId)) }}
```

Comment:
I think you can use type.s3 at the first level and render this if those are set and the irsa one if they are not?